### DISCUSSION TITLE
Device Mode stopped working on Win/MacOS

### DISCUSSION THREAD

**Discussion by Absynth723** [geerlingguy/pi-webcam#46] [Q&A] [answered]
I've been using pi-webcam to great success, but all of a sudden between two power cycles, device mode stopped working - the pi is not visible via usb anymore. I had enabled SSH and Wifi so I could check on the Pi, and all services regarding the cam seemed to be running. Only, it didn't show up on the USB device manager on any of my machines.
So I reflashed with a known-good image of pi-webcam (which I had made after an unsafe shutdown trashed the OS once). Still not working. I just re-re-flashed with the same image, and (duh) it's stll not working.
Is there anything I can do to debug this? I know my way around linux and the Pi, I'm just not sure which kind of magic you do to enable the USB device mode.

**Comment by geerlingguy** [comment]
Have you recently updated Raspberry Pi OS? If so, maybe it's a bug in the latest firmware... I know with the Pi 4 there was a release last year that messed things up, and the only workaround was to use an older OS release version.

**Reply by Absynth723** [reply]
Hi Jeff, 
I don't think that I did. 
I made a known-working image and reflashed that. FWIW, here's my versions:

pi@raspberrypi:~ $ vcgencmd version
Feb 25 2021 12:13:12
Copyright (c) 2012 Broadcom
version 564e5f9b852b23a330b1764bcf0b2d022a20afd0 (clean) (release) (start_x)

pi@raspberrypi:~ $ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster

I'll do a clean reinstall this morning and see if the situation resolves itself.

**Comment by Absynth723** [comment, marked as answer]
OK, so I reinstalled on a freshly flashed raspbian (I use a RPi4 to do the install, then swap the MicroSD to the ZeroW) and it works again. I have no idea what happened there. 
So, I guess this is kinda resolved. Anyhoo, how can one debug this issue properly, do you have some kind of workflow for that? Or is it generally a case of "sod it, just re-flash and be done with it"? I have to say, I find this whole device mode thing intriguing, yet mysterious, and it kind of stirs my curiosity.
